libnmstate/nm/connection.py

```diff
@@ -324,6 +324,18 @@ def _activation_progress_check(self, action):
             )
             self._activation_clean_up()
             self._ctx.finish_async(action)
+        elif (
+            not self._is_activating()
+            and self._is_sriov_parameter_not_supported_by_driver()
+        ):
+            reason = (
+                f"The device={self.devname} does not support one or "
+                "more of the SR-IOV parameters set."
+            )
+            self._activation_clean_up()
+            self._ctx.fail(
+                NmstateLibnmError(f"{action} failed: reason={reason}")
```

Comment:
Should be `NmstateValueError` as desire state is invalid for this SRIOV NIC.

Reply:
Done, thanks!